fitnessllm_dataplatform/infrastructure/RedisConnect.py

```
@@ -0,0 +1,106 @@
+        """
+        self.open_connection()
+        try:
+            value = self.interface.get(key)
+
+            if value is None:
+                raise redis.RedisError(f"Failed to get key '{key}'; does not exist")
+            else:
+                return json.loads(value)
+
+        except redis.RedisError as exc:
+            raise redis.RedisError(f"Failed to get key '{key}': {exc}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)